actors/init/src/main/java/org/sunbird/init/actors/SystemInitActor.java

```diff
@@ -0,0 +1,274 @@
+    Map<String, Object> filters = new HashMap<>();
+    filters.put(JsonKey.IS_ROOT_ORG, true);
+    Map<String, Object> esResult =
+        elasticSearchComplexSearch(
+            filters, EsIndex.sunbird.getIndexName(), EsType.organisation.getTypeName());
+    if (isNotNull(esResult)
+        && esResult.containsKey(JsonKey.CONTENT)
+        && isNotNull(esResult.get(JsonKey.CONTENT))
+        && ((List) esResult.get(JsonKey.CONTENT)).size() > 0) {
+      return true;
+    }
+    return false;
+  }
+
+  /**
+ * This method creates rootOrg if it is not exists already
+ * 
+ * @param actorMessage instance of Request class with rootOrg data.
+ */
+  private void checkAndCreateRootOrg(Request actorMessage) {
+    if (isRootOrgExists() == false) {
+      Map<String, Object> req =
+          (Map<String, Object>) actorMessage.getRequest().get(JsonKey.ORGANISATION);
+      Util.DbInfo orgDbInfo = Util.dbInfoMap.get(JsonKey.ORG_DB);
+      validateChannelUniqueness(req);
+      req.put(JsonKey.CREATED_BY, JsonKey.INITIALISER);
+      String uniqueId = ProjectUtil.getUniqueIdFromTimestamp(actorMessage.getEnv());
+      req.put(JsonKey.ID, uniqueId);
+      req.put(JsonKey.ROOT_ORG_ID, uniqueId);
+      req.put(JsonKey.HASHTAGID, uniqueId);
+      req.put(JsonKey.CREATED_DATE, ProjectUtil.getFormattedDate());
+      req.put(JsonKey.STATUS, ProjectUtil.OrgStatus.ACTIVE.getValue());
+      req.put(JsonKey.IS_ROOT_ORG, true);
+      req.put(JsonKey.IS_DEFAULT, true);
+
+      String slug = Slug.makeSlug((String) req.get(JsonKey.CHANNEL), true);
+      if (isSlugUnique(slug)) {
+        req.put(JsonKey.SLUG, slug);
+      } else {
+        ProjectCommonException.throwClientErrorException(
+            ResponseCode.slugIsNotUnique, ResponseCode.slugIsNotUnique.getErrorMessage());
+      }
+
+      boolean isChannelRegistered = Util.registerChannel(req);
+      if (!isChannelRegistered) {
+        ProjectCommonException.throwServerErrorException(
+            ResponseCode.channelRegFailed, ResponseCode.channelRegFailed.getErrorMessage());
+      }
+
+      Organization org = mapper.convertValue(req, Organization.class);
+      req = mapper.convertValue(org, Map.class);
+      Response result =
+          cassandraOperation.insertRecord(orgDbInfo.getKeySpace(), orgDbInfo.getTableName(), req);
+      ProjectLogger.log("Org data saved into cassandra.");
+      ProjectLogger.log("Created org id is ----." + uniqueId);
+      result.getResult().put(JsonKey.ORGANISATION_ID, uniqueId);
+      sender().tell(result, self());
+      Request orgIndexReq = new Request();
+      orgIndexReq.getRequest().put(JsonKey.ORGANISATION, req);
+      orgIndexReq.setOperation(ActorOperations.INSERT_ORG_INFO_ELASTIC.getValue());
+      ProjectLogger.log("Calling background job to save org data into ES" + uniqueId);
+      tellToAnother(orgIndexReq);
+    } else {
+      setInitialisationStatus(JsonKey.COMPLETED);
+      ProjectCommonException.throwClientErrorException(
```

Comment:
I'm not sure that this should return an error. Possibly should return success?

Reply:
As discussed earlier,If rootOrg already exists we should update isRootOrgInitialised to 'completed' and we should throw error.should we change it?